### ISSUE TITLE
Sync DropBox failed

### ISSUE THREAD

**Issue by Iraecio** [sebastianfeldmann/phpbu#130]
Exception 'phpbu\App\Backup\Sync\Exception' with message 'Error in call to API function "files/upload": HTTP header "Dropbox-API-Arg": path: 'development/mysql-20180204-2253.sql.bz2' did not match pattern '(/(.|[\r\n])*)|(ns:[0-9]+(/.*)?)|(id:.*)''
in phar://......./phpbu.phar/Backup/Sync/Dropbox.php:87

**Comment by sebastianfeldmann** [comment]
Which phpbu version are you unsing?

**Comment by Iraecio** [comment]
5.0.9

**Comment by sebastianfeldmann** [comment]
Looks like the Dropbox API is throwing an exception :/
Can you confirm the backup is indeed created and the path is correct?
I’ll investigate this further as soon as I’m back on my desk

**Comment by Iraecio** [comment]
![sem titulo](https://user-images.githubusercontent.com/8146933/35794539-e6ab3024-0a34-11e8-9768-c4acdefb8c72.png)

Exception 'phpbu\App\Backup\Sync\Exception' with message 'Error in call to API function "files/upload": HTTP header "Dropbox-API-Arg": path: 'backups/mysql-20180205-0020.sql.bz2' did not match pattern '(/(.|[\r\n])*)|(ns:[0-9]+(/.*)?)|(id:.*)''
in phar://C:/Windows/AppCompat/nginx/apps/api/phpbu.phar/Backup/Sync/Dropbox.php:87

**Comment by sebastianfeldmann** [comment]
Can you share your config file?
Off course, please replace all sensitive data with ”*”.
Thank you ;)

**Comment by Iraecio** [comment]
same of example with only backups source and syncs dropbox

**Comment by Iraecio** [comment]
`{
  "verbose": true,
  "logging": [
    {
      "type": "json",
      "target": "data\/logs\/backup.json"
    }
  ],
  "backups": [
    {
      "name": "DataBase",
      "source": {
        "type": "mysqldump",
        "options": {
          "databases": "development",
          "user": "*********",
          "password": "***********",
          "pathToMysqldump": "********/mysql\/bin"
        }
      },
      "target": {
        "dirname": "data\/backup",
        "filename": "mysql-%Y%m%d-%H%i.sql",
        "compress": "bzip2"
      },
      "syncs": [
        {
          "type": "dropbox",
          "options": {
            "token": "**************************",
            "path": "backups"
          }
        }
      ]
    }
  ]
}`

**Comment by sebastianfeldmann** [comment]
Sorry it took a while, but I finally could reproduce your error.

The Problem is, there is a slash missing in the dropbox configuration.
In your configuration is states "backups".
It should be

```json
{ "type": "dropbox", "options": { "token": "***", "path": "/backups" }
```

That did the trick for me.
I'll add a option validator for this, so there will be a nice exception message.

**Comment by sebastianfeldmann** [comment]
I fact I decided to not throw an Exception. Instead I make sure that the Dropbox path always contains a leading /.
If not provided by the configuration I will add it manually.
Fix will be included in 5.1.0

**Comment by Iraecio** [comment]
Thx. this Work
